devicemodel/hw/platform/uart.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2018-2024 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <errno.h>
+
+#include "acpi.h"
+#include "vmmapi.h"
+#include "log.h"
+#include "acpi_dev.h"
+#include "dm.h"
+
+static uint32_t uart_idx = 0U;
+
+static int pt_uart_init(struct vmctx *ctx, struct acrn_acpidev *uart_dev)
+{
+	if(!ctx || !uart_dev)
+		return -EINVAL;
+
+	strncpy(uart_dev->name, "uart", sizeof(uart_dev->name));
+	return init_pt_acpidev(ctx, uart_dev);
+}
+
+static int is_hid_legacy_uart(char *opts)
+{
+	if (!opts || !*opts)
+		return false;
+
+	int ret = true;
+	char *hid =NULL;
+	char *vtopts = NULL, *bak_vtopts = NULL;
+
+	vtopts = strdup(opts);
+	bak_vtopts = vtopts;
+	hid = strsep(&vtopts, ",");
+	if (strstr(hid, "PNP0501") == NULL)
+		ret = false;
+
+	free(bak_vtopts);
+	return ret;
+}
+
+static void uart_write_dsdt(struct vmctx *ctx, struct acpi_dev *acpidev)
+{
+	if (!ctx || !acpidev)
+		return;
+	struct acrn_acpidev *dev = &acpidev->dev;
+	struct acrn_acpires *res =NULL;
+	
+	dsdt_line("");
+	dsdt_line("Device (COM%d)", uart_idx);
```

Comment:
Will it have the same name as emulated COM in LPC?

Reply:
Passthrough UART may have same name as emulated COM in LPC. However, they are under different ACPI node. Emulated COM in LPC is under \_SB_.PCI0.ISA_ node while passthrough UART is under root node. So COM devices with same name won't cause any conflict during system boot.